coderd/database/queries/provisionerdaemons.sql

```diff
@@ -28,6 +28,54 @@ JOIN
 WHERE
     provisioner_jobs.id = ANY(@provisioner_job_ids :: uuid[]);
 
+-- name: GetProvisionerDaemonsWithStatusByOrganization :many
+SELECT
+	sqlc.embed(pd),
+	CASE
+		WHEN pd.last_seen_at IS NULL OR pd.last_seen_at < (NOW() - (@stale_interval_ms::bigint || ' ms')::interval)
+		THEN 'offline'
+		ELSE CASE
+			WHEN current_job.id IS NOT NULL THEN 'busy'
+			ELSE 'idle'
+		END
+	END::provisioner_daemon_status AS status,
+	pk.name AS key_name,
+	-- NOTE(mafredri): sqlc.embed doesn't support nullable tables nor renaming them.
+	current_job.id AS current_job_id,
+	current_job.job_status AS current_job_status,
+	previous_job.id AS previous_job_id,
+	previous_job.job_status AS previous_job_status
+FROM
+	provisioner_daemons pd
+JOIN
+	provisioner_keys pk ON pk.id = pd.key_id
+LEFT JOIN
```

Comment:
nit: I'm curious if you can improve readability by adopting CTEs

```
WITH previous_jobs AS (
...
),
current_jobs AS (
...
)
SELECT ...
`

Reply:
Personally I find the left join more readable since it's quite simple (single table). I typically love CTEs but they have a few drawbacks like not having indexes on the CTE (can be slower sometimes but requires more data than here) and needing to re-apply WHERE filters (if applicable).